server/modules/navigator/navigator.go

```diff
@@ -0,0 +1,370 @@
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/apex/log"
+	"github.com/security-onion-solutions/securityonion-soc/model"
+	"github.com/security-onion-solutions/securityonion-soc/module"
+	"github.com/security-onion-solutions/securityonion-soc/server"
+	"gopkg.in/yaml.v3"
+)
+
+const DEFAULT_INTERVAL_MINUTES = 60
+const DEFAULT_OUTPUT_PATH = "/opt/sensoroni/navigator"
+const DEFAULT_LOOKBACK_DAYS = 7
+const ALERT_EXTRACTION_TIMEOUT = 3 * time.Minute
+
+var (
+	// suricataTechniqueRegex matches MITRE technique IDs in Suricata rules
+	suricataTechniqueRegex = regexp.MustCompile(`mitre_technique_id\s+([^,;\s]+)`)
+	// sigmaTechniqueRegex matches MITRE technique IDs in Sigma rules
+	sigmaTechniqueRegex = regexp.MustCompile(`(?i)t\d{4}(?:\.\d{3})?`)
+)
+
+// techniqueMap is a helper type for storing unique technique IDs
+type techniqueMap map[string]struct{}
+
+// navigatorLayer represents the structure for generating ATT&CK Navigator layers
+type navigatorLayer struct {
+	name       string
+	techniques techniqueMap
+}
+
+// layerConfig represents the configuration for a navigator layer including its output path
+type layerConfig struct {
+	name       string
+	techniques techniqueMap
+	filePath   string
+}
+
+type Navigator struct {
+	config       module.ModuleConfig
+	server       *server.Server
+	interval     time.Duration
+	outputPath   string
+	lookbackDays int
+	stopChan     chan bool
+	techniques   techniqueMap
+}
+
+func NewNavigator(srv *server.Server) *Navigator {
+	return &Navigator{
+		server:     srv,
+		techniques: make(techniqueMap),
+		stopChan:   make(chan bool),
+	}
+}
+
+func (nav *Navigator) PrerequisiteModules() []string {
+	return nil
+}
+
+func (nav *Navigator) Init(cfg module.ModuleConfig) error {
+	nav.config = cfg
+	intervalMinutes := module.GetIntDefault(cfg, "intervalMinutes", DEFAULT_INTERVAL_MINUTES)
+	nav.interval = time.Duration(intervalMinutes) * time.Minute
+	nav.outputPath = module.GetStringDefault(cfg, "outputPath", DEFAULT_OUTPUT_PATH)
+	nav.lookbackDays = module.GetIntDefault(cfg, "lookbackDays", DEFAULT_LOOKBACK_DAYS)
+	return nil
+}
+
+func (nav *Navigator) Start() error {
+	go nav.run()
+	return nil
+}
+
+func (nav *Navigator) Stop() error {
+	nav.stopChan <- true
+	return nil
+}
+
+func (nav *Navigator) IsRunning() bool {
+	return nav.stopChan != nil
+}
+
+func (nav *Navigator) run() {
+	ticker := time.NewTicker(nav.interval)
+	defer ticker.Stop()
+
+	logger := log.WithField("component", "navigator")
+
+	// Generate initial set of layers
+	if err := nav.generateNavigatorLayer(nav.server.Context, logger); err != nil {
+		logger.WithError(err).Error("Failed to generate initial navigator layers")
+	}
+
+	for {
+		select {
+		case <-nav.stopChan:
+			return
+		case <-ticker.C:
+			if err := nav.generateNavigatorLayer(nav.server.Context, logger); err != nil {
+				logger.WithError(err).Error("Failed to generate navigator layers")
+			}
+		}
+	}
```

Comment:
It's good practice to log an Info line when exiting a long-running loop function. Helps for troubleshooting, in case the loop crashed or got stuck on a query call.